src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/CaseNoteFilter.java

```diff
@@ -59,9 +52,38 @@ public Predicate toPredicate(@NotNull final Root<OffenderCaseNote> root, @NotNul
             predicateBuilder.add(cb.lessThan(root.get("occurrenceDate"), endDate.plusDays(1)));
         }
 
+        if(types!=null&&!types.isEmpty()){
+           predicateBuilder.add(getTypesPredicate(root,cb));
+        }
+
         final var predicates = predicateBuilder.build();
         return cb.and(predicates.toArray(new Predicate[0]));
+    }
+
+    private Predicate getTypesPredicate(final Root<OffenderCaseNote> root, final CriteriaBuilder cb) {
+        final ImmutableList.Builder<Predicate> typesPredicateorBuilder = ImmutableList.builder();
+        types.forEach((String type, List<String> subTypes)->{
+
+            if(subTypes==null||subTypes.isEmpty()){
```

Comment:
having to work in java is painful - don't think `subTypes` can ever be `null`?

Reply:
Removed unnecessary null check